Orgs

blomquisg opened the pull request:
Update settings.yaml.example
  -- added comments
Use unscoped query in installation media db migration
Fix create_domains permission
Fix select_organization default data in loader
Force users in no org to see no organizationally bound objects
  -- fixed org_ids calculation, and added a comment explaining why org_ids are gathered that way

blomquisg committed:
Adds Organizations Object

adding setup_user method in test_helper
Add debug support to tests
Allow admin to add users to organizations and organizations to users
Add organization-smart_proxy relationship
change to_s on user to include username
Fix user form so admins cannot be assigned to orgs
Add a org-user test
Add org-compute_resource relationship
Use shortcut when mapping to smart_proxy name in to_s method
Add org-media relationship
Fix some test failures
Add domain and hostgroup associations with organizations
Add org switcher and "current" organization
  * need to test "default_data/loader.rb"
Add org scope to filter org-restricted objects by the current selected org for non-admin users
Set default (current) organization in single-org mode
Add show_organization_tab? to only show organization functionality in the UI when orgnizaitons are enabled
Fix up configurations for org modes
  * allow for only one org mode at a time (single or multi)
  * setup routes only when orgs are enabled
Allow lambdas for default_scope (fixed in rails 3.1)
  * monkey patch active record to allow for lambdas in default scope
    - https://rails.lighthouseapp.com/projects/8994/tickets/1812-default
    - this implementation does not allow for multiple calls to default scope, so
      all calls need to be collapsed into the same lambda
  * add apply_org_scope class method to organization model to easily scope on
    organization relationship
  * change all default_scope invocations to lambdas that hand off to
    organization class method for runtime scoping
Add environment-organization relationship
to multiple orgs to select multiple orgs when saving an environment ... just to
show difference between the mode with the other objects, which saves the object
against only the currently selected org in single-org mode
Disable the ability for an admin to accidentally select a 'current org'

blomquisg committed:
Adding new org settings to example yaml

blomquisg committed:
Updating existing installation media migration

This fails on new DB setup with the org changes to default_scope

blomquisg committed:
Changing create_domain permission to create_domains so the permission checking algorithm finds it under the controller name

blomquisg committed:
Fixing permission name for select_organizations so Viewer can select an org from the drop down

blomquisg committed:
Changing apply_org_scope to hide all objects from users in no org.

If organizations are enabled (multi or single), then users with no organization
will see no objects in (or not in) orgs.

ohadlevy commented on app/controllers/compute_resources_controller.rb:
```diff
@@ -32,6 +32,11 @@ def show
+        if SETTINGS[:single_org]
+          @compute_resource.organization_ids = [Organization.current.id]
+        end
+      end
```
does it make sense to move it into a common helper method?

also, not 100% sure, but does it make sense to move it to the model (as both User.current and Settings are avail)

blomquisg replied:
On Tue, Jul 31, 2012 at 2:09 AM, Ohad Levy <
reply@reply.github.com

> wrote:
> 
> > @@ -32,6 +32,11 @@ def show
> >    def create
> >      if params[:compute_resource].present? &&
> > params[:compute_resource][:provider].present?
> >        @compute_resource = ComputeResource.new_provider
> > params[:compute_resource]
> > -      unless User.current.admin?
> > -        if SETTINGS[:single_org]
> > -          @compute_resource.organization_ids = [Organization.current.id
> >   ]
> > -        end
> > -      end
> 
> does it make sense to move it into a common helper method?
> 
> also, not 100% sure, but does it make sense to move it to the model (as
> both User.current and Settings are avail)

Interesting.  Something like:

def Organization.when_single_org &block
  unless User.current.admin?
    if SETTINGS[:single_org]
      yield
    end
  end
end

I just need a little help on the rubiness of it, I'm sure.

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/theforeman/foreman/pull/121/files#r1270450

ohadlevy commented on app/views/compute_resources/_form.html.erb:
```diff
@@ -1,12 +1,36 @@
+
+  <% else %>
+    <%= text_f f, :name %>
+    <%= selectable_f f, :provider, ComputeResource::PROVIDERS, { :include_blank => "Choose a provider"},
```
can we use the if twice (e.g. for both tab and tab content) instead of duplicating the other tabs view?

blomquisg replied:
On Tue, Jul 31, 2012 at 2:12 AM, Ohad Levy <
reply@reply.github.com

> wrote:
> 
> > -        <%= text_f f, :name %>
> > -        <%= selectable_f f, :provider, ComputeResource::PROVIDERS, {
> >   :include_blank => "Choose a provider"},
> > -                         {:disabled=> f.object.uuid.present?,
> >   :'data-url'=> provider_selected_compute_resources_path, :onchange =>
> >   'providerSelected(this);'} %>
> > -        <%= textarea_f f, :description, :rows => 3 %>
> > -        <div id='compute_connection'>
> > -          <%= render
> >   "compute_resources/form/#{@compute_resource.provider.downcase}", :f => f
> >   unless @compute_resource.provider.empty? %>
> > -        </div>
> > -      </div>
> > -      <div class="tab-pane" id="organizations">
> > -        <%= multiple_checkboxes f, :organizations, @compute_resource,
> >   Organization %>
> > -      </div>
> > -    </div>
> >   +
> > -  <% else %>
> > -    <%= text_f f, :name %>
> > -    <%= selectable_f f, :provider, ComputeResource::PROVIDERS, {
> >   :include_blank => "Choose a provider"},
> 
> can we use the if twice (e.g. for both tab and tab content) instead of
> duplicating the other tabs view?

Yeah, I'm sure that'll work.

This was admittedly some haphazard coding due to a last minute change...

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/theforeman/foreman/pull/121/files#r1270458

blomquisg committed:
move common single_org checking code to organization model to be reused in controllers

blomquisg committed:
clean up views that use 'show_organization_tab?' helper method

blomquisg committed:
Change apply_org_scope to with_org_scope

The Organization model now provides 'with_org_scope' method, which allows other
domain objects to provide their own base scoping to be further scoped to
organizations.

blomquisg committed:
Adding host-org mapping

Commit:
Added puppetclasses to org scope

Commit:
Show the organization tab based on the show_organization_tab method

blomquisg committed:
Merge remote-tracking branch 'blomquisg/orgs' into orgs

blomquisg committed:
Merge pull request #1 from skottler/orgs_puppetclasses

blomquisg committed:
Update existing host migrations to use unscoped

Now that hosts are organizationally bound and use the
Organization.with_org_scope method for default_scope, two existing host
migrations need to select Hosts unscoped in order to avoid trying to determine
if the current user is an admin.

skottler commented:
Since the changeset has diverged so much from the content of the pull request, this should get closed out.
